ledger/heavy/executor/backuper.go

```diff
@@ -0,0 +1,293 @@
+}
+
+func checkConfig(config configuration.Backup) error {
+	if err := isPathExists(config.TmpDirectory); err != nil {
+		return errors.Wrap(err, "checkDirectory returns error")
+	}
+	if err := isPathExists(config.TargetDirectory); err != nil {
+		return errors.Wrap(err, "checkDirectory returns error")
+	}
+	if len(config.ConfirmFile) == 0 {
+		return errors.New("ConfirmFile can't be empty")
+	}
+	if len(config.MetaInfoFile) == 0 {
+		return errors.New("MetaInfoFile can't be empty")
+	}
+	if len(config.DirNameTemplate) == 0 {
+		return errors.New("DirNameTemplate can't be empty")
+	}
+	if config.BackupWaitPeriod == 0 {
+		return errors.New("BackupWaitPeriod can't be 0")
+	}
+	if len(config.BackupFile) == 0 {
+		return errors.New("BackupFile can't be empty")
+	}
+
+	return nil
+}
+
+func NewBackupMaker(ctx context.Context, backuper store.Backuper, config configuration.Backup, lastBackupedPulse insolar.PulseNumber) (*BackupMakerDefault, error) {
+	if config.Enabled {
+		if err := checkConfig(config); err != nil {
+			return nil, errors.Wrap(err, "bad config")
+		}
+	} else {
+		inslogger.FromContext(ctx).Info("Backup is disabled")
+	}
+
+	return &BackupMakerDefault{
+		backuper:          backuper,
+		config:            config,
+		lastBackupedPulse: lastBackupedPulse,
+	}, nil
+}
+
+func move(ctx context.Context, what string, toDirectory string) error {
+	inslogger.FromContext(ctx).Debugf("backuper. move %s -> %s", what, toDirectory)
+	err := os.Rename(what, toDirectory)
+	if err != nil {
+		return errors.Wrapf(err, "can't move %s to %s", what, toDirectory)
+	}
+
+	return nil
+}
+
+// waitForBackup waits for file filePath appearance
+func waitForBackup(ctx context.Context, filePath string, numIterations uint) error {
+	inslogger.FromContext(ctx).Debug("waiting for ", filePath)
+	for i := uint(0); i < numIterations; i++ {
+		if err := isPathExists(filePath); err != nil {
+			if os.IsNotExist(err) {
+				inslogger.FromContext(ctx).Debugf("backup confirmation ( %s ) still doesn't exists. Sleep second.", filePath)
+				time.Sleep(time.Second)
+				continue
+			}
+			return errors.Wrap(err, "isPathExists return error")
+		}
+		return nil
+	}
+
+	return errors.New("no backup confirmation for pulse")
+}
+
+func writeBackupInfoFile(hash string, pulse insolar.PulseNumber, since uint64, upto uint64, to string) error {
+	bi := BackupInfo{SHA256: hash, Pulse: pulse, LastBackupedVersion: upto, Since: since}
+
+	rawInfo, err := json.MarshalIndent(bi, "", "    ")
+	if err != nil {
+		return errors.Wrap(err, "can't marshal backup info")
+	}
+
+	err = ioutil.WriteFile(to, rawInfo, 0600)
+	return errors.Wrapf(err, "can't write file %s", to)
+}
+
+func calculateFileHash(f *os.File) (string, error) {
+	hasher := sha256.New()
+	if _, err := io.Copy(hasher, f); err != nil {
+		return "", errors.Wrap(err, "io.Copy return error")
+	}
+	return fmt.Sprintf("%x", hasher.Sum(nil)), nil
+}
+
+// prepareBackup make incremental backup and write auxiliary file with meta info
```

Comment:
add returned uint64 value description in godoc

Reply:
It's not an exporting method. But, no problem